python/ray/util/spark/utils.py

```diff
@@ -194,13 +197,22 @@ def calc_mem_ray_head_node(configured_object_store_bytes):
     else:
         available_physical_mem = psutil.virtual_memory().total
 
+    available_physical_mem = (
+        available_physical_mem * _RAY_ON_SPARK_NODE_MEMORY_BUFFER_OFFSET
+    )
+
     if RAY_ON_SPARK_DRIVER_SHARED_MEMORY_BYTES in os.environ:
         available_shared_mem = int(os.environ[RAY_ON_SPARK_DRIVER_SHARED_MEMORY_BYTES])
     else:
         available_shared_mem = psutil.virtual_memory().total
```

Comment:
@WeichenXu123 This looks like the Physical memory? Should this be something like `shutil.disk_usage("/dev/shm").total` like what we do for [worker shared memory](https://github.com/ray-project/ray/blob/4b8161d1093b93fdf26ee77ef6cea6b2cc86927e/python/ray/util/spark/utils.py#L177)?